### DISCUSSION TITLE
Сводная таблица

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#181] [Как это сделать?] [answered]
[Сводные таблицы.xlsx](https://github.com/totumonline/totum-ru-issues-and-discussions/files/9268655/default.xlsx)
Таблица 1 у нас есть, мы хотим получить таблицы 2 и 3.
В Excel это делается через Сводные таблицы. Хотим понять какой аналогичный механизм есть в totum.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
### Через автозаполнение временной таблицы:

Источник:

<img width="477" alt="Снимок экрана 2022-08-05 в 18 39 48" src="https://user-images.githubusercontent.com/55755565/183112375-ed0b3b97-7156-4184-815e-acfc02a76476.png">

Временная таблица:

<img width="384" alt="Снимок экрана 2022-08-05 в 18 40 02" src="https://user-images.githubusercontent.com/55755565/183112403-7f1e31d9-b097-434f-bf01-59a59f5d7dc0.png">

Код в `h_insert` (не факт, что здесь самый изящный способ генерации последовательного списка):

```
= : listReplace(list: $repeat; action: $val; key: "key")
	val: $#key + 1

repeat: listRepeat(item: 1; num: $max)

	max: listMax(list: $attempts; default: 1)

	attempts: listAdd(list: $atp[[first_attempt]]; list: $atp[[second_attempt]])
		~atp: selectRowList(table: 'primer_na_gh'; field: 'first_attempt'; field: 'second_attempt')
```

Код в `books_in_first`:

```
= : listCount(list: $list)

	list: selectList(table: 'primer_na_gh'; field: 'id'; where: 'first_attempt' = #insert)
```

**Reply by Makhmut2445** [reply]
![image](https://user-images.githubusercontent.com/103171127/183367004-b891c931-d60c-45cb-9778-651d12444bef.png)
В таблице с списком книг есть поле ответственный... Как в временной таблице сделать по ним разбивку по дереву (Тип - По другой таблице)?

**Reply by ttm-alexeypolunin** [reply]
Структура дерева по Ответственному где-то определена? Если нет, то это делается через генерацию Временной таблицы с деревом в фоне.
